TS Server fatal error:  Debug Failure. Text of property name cannot be read from non-literal-valued ComputedPropertyNames

**TypeScript Version:** 3.8.3
**Steps to reproduce crash**
1.
2.
3.
**TS Server Error Stack**
```
Error: Debug Failure. Text of property name cannot be read from non-literal-valued ComputedPropertyNames
    at Object.getTextOfPropertyName (tsserver.js:13264:33)
    at _loop_3 (tsserver.js:27326:36)
    at convertObjectLiteralExpressionToJson (tsserver.js:27368:17)
    at convertPropertyValueToJson (tsserver.js:27427:32)
    at Object.convertToObjectWorker (tsserver.js:27309:16)
    at Object.parseSourceFile (tsserver.js:18942:20)
    at Object.createSourceFile (tsserver.js:18772:29)
    at Object.createLanguageServiceSourceFile (tsserver.js:134984:29)
    at acquireOrUpdateDocument (tsserver.js:112201:37)
    at Object.acquireDocumentWithKey (tsserver.js:112174:20)
    at getOrCreateSourceFileByPath (tsserver.js:135301:41)
    at Object.getOrCreateSourceFile [as getSourceFile] (tsserver.js:135253:24)
    at findSourceFile (tsserver.js:99143:29)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at processImportedModules (tsserver.js:99484:25)
    at findSourceFile (tsserver.js:99191:17)
    at tsserver.js:99020:85
    at getSourceFileFromReferenceWorker (tsserver.js:98987:34)
    at processSourceFile (tsserver.js:99020:13)
    at processRootFile (tsserver.js:98843:13)
    at tsserver.js:97809:60
    at Object.forEach (tsserver.js:167:30)
    at Object.createProgram (tsserver.js:97809:16)
    at synchronizeHostData (tsserver.js:135218:26)
    at Proxy.getProgram (tsserver.js:135310:13)
    at InferredProject.Project.updateGraphWorker (tsserver.js:139786:53)
    at InferredProject.Project.updateGraph (tsserver.js:139729:42)
    at ProjectService.assignOrphanScriptInfoToInferredProject (tsserver.js:141705:25)
    at ProjectService.assignProjectToOpenedScriptInfo (tsserver.js:143132:26)
    at tsserver.js:143396:107
    at Object.mapDefined (tsserver.js:562:30)
    at ProjectService.applyChangesInOpenFiles (tsserver.js:143396:48)
    at Session.handlers.ts.createMapFromTemplate._a.<computed> (tsserver.js:144260:46)
    at tsserver.js:146003:88
    at IOSession.Session.executeWithRequestId (tsserver.js:145994:28)
    at IOSession.Session.executeCommand (tsserver.js:146003:33)
    at IOSession.Session.onMessage (tsserver.js:146027:35)
    at Interface.<anonymous> (tsserver.js:147342:27)
    at Interface.emit (events.js:203:13)
    at Interface._onLine (readline.js:316:10)
    at Interface._normalWrite (readline.js:461:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
```

Hello, I also have this problem when I import a JSON file and I don't know how to solve it...

Angular component :
```javascript
import { Component } from "@angular/core";
import * as data from "../../app/nexus-data.json";

@Component({
  selector: "app-deck-update",
  templateUrl: "./deck-update.component.html",
  styleUrls: ["./deck-update.component.scss"],
})
export class DeckUpdateComponent {
  nexusApps: any = (data as any).default;
}
```

JSON file : 
```javascript
[
  {
    "version": "19.00",
    "name": "7-Zip",
    "url": "http://www.7-zip.org"
  },
  {
    "version": "4.0.28",
    "name": "Balsamiq",
    "url": "https://balsamiq.com/"
  }
]
```


NPM console :
```batch
2 import * as data from "../../app/nexus-data.json";
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/app/deck-update/deck-update.component.ts:2:23 - error TS5012:  
Cannot read file 'C:/Users/onezime.j/Desktop/synchronaut/src/app/nexus-data.json':  
Debug Failure. Text of property name cannot be read from non-literal-valued 
ComputedPropertyNames.
```

Thanks for your help, it drives me crazy !!

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -7192,10 +7192,6 @@ namespace ts {
             return symbol && getSymbolLinks(symbol).type || getTypeForVariableLikeDeclaration(node, /*includeOptionality*/ false);
         }
 
-        function isComputedNonLiteralName(name: PropertyName): boolean {
-            return name.kind === SyntaxKind.ComputedPropertyName && !isStringOrNumericLiteralLike(name.expression);
-        }
-
         function getRestType(source: Type, properties: PropertyName[], symbol: Symbol | undefined): Type {
             source = filterType(source, t => !(t.flags & TypeFlags.Nullable));
             if (source.flags & TypeFlags.Never) {

--- a/src/compiler/commandLineParser.ts
+++ b/src/compiler/commandLineParser.ts
@@ -1721,7 +1721,7 @@ namespace ts {
                     errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element.name, Diagnostics.String_literal_with_double_quotes_expected));
                 }
 
-                const textOfKey = getTextOfPropertyName(element.name);
+                const textOfKey = isComputedNonLiteralName(element.name) ? undefined : getTextOfPropertyName(element.name);
                 const keyText = textOfKey && unescapeLeadingUnderscores(textOfKey);
                 const option = keyText && knownOptions ? knownOptions.get(keyText) : undefined;
                 if (keyText && extraKeyDiagnostics && !option) {

--- a/src/compiler/utilities.ts
+++ b/src/compiler/utilities.ts
@@ -876,6 +876,10 @@ namespace ts {
         return info.declaration ? declarationNameToString(info.declaration.parameters[0].name) : undefined;
     }
 
+    export function isComputedNonLiteralName(name: PropertyName): boolean {
+        return name.kind === SyntaxKind.ComputedPropertyName && !isStringOrNumericLiteralLike(name.expression);
+    }
+
     export function getTextOfPropertyName(name: PropertyName | NoSubstitutionTemplateLiteral): __String {
         switch (name.kind) {
             case SyntaxKind.Identifier:

--- a/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.errors.txt
+++ b/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.errors.txt
@@ -1,0 +1,17 @@
+tests/cases/compiler/b.json(2,5): error TS1327: String literal with double quotes expected.
+
+
+==== tests/cases/compiler/file1.ts (0 errors) ====
+    import b1 = require('./b.json');
+    let x = b1;
+    import b2 = require('./b.json');
+    if (x) {
+        x = b2;
+    }
+    
+==== tests/cases/compiler/b.json (1 errors) ====
+    {
+        [a]: 10
+        ~~~
+!!! error TS1327: String literal with double quotes expected.
+    }

--- a/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.js
+++ b/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.js
@@ -1,0 +1,29 @@
+//// [tests/cases/compiler/requireOfJsonFileWithComputedPropertyName.ts] ////
+
+//// [file1.ts]
+import b1 = require('./b.json');
+let x = b1;
+import b2 = require('./b.json');
+if (x) {
+    x = b2;
+}
+
+//// [b.json]
+{
+    [a]: 10
+}
+
+//// [b.json]
+var _a;
+_a = {},
+    _a[a] = 10,
+    _a;
+//// [file1.js]
+"use strict";
+exports.__esModule = true;
+var b1 = require("./b.json");
+var x = b1;
+var b2 = require("./b.json");
+if (x) {
+    x = b2;
+}

--- a/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.symbols
+++ b/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.symbols
@@ -1,0 +1,24 @@
+=== tests/cases/compiler/file1.ts ===
+import b1 = require('./b.json');
+>b1 : Symbol(b1, Decl(file1.ts, 0, 0))
+
+let x = b1;
+>x : Symbol(x, Decl(file1.ts, 1, 3))
+>b1 : Symbol(b1, Decl(file1.ts, 0, 0))
+
+import b2 = require('./b.json');
+>b2 : Symbol(b2, Decl(file1.ts, 1, 11))
+
+if (x) {
+>x : Symbol(x, Decl(file1.ts, 1, 3))
+
+    x = b2;
+>x : Symbol(x, Decl(file1.ts, 1, 3))
+>b2 : Symbol(b2, Decl(file1.ts, 1, 11))
+}
+
+=== tests/cases/compiler/b.json ===
+{
+    [a]: 10
+>[a] : Symbol([a], Decl(b.json, 0, 1))
+}

--- a/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.types
+++ b/tests/baselines/reference/requireOfJsonFileWithComputedPropertyName.types
@@ -1,0 +1,29 @@
+=== tests/cases/compiler/file1.ts ===
+import b1 = require('./b.json');
+>b1 : { [x: number]: number; }
+
+let x = b1;
+>x : { [x: number]: number; }
+>b1 : { [x: number]: number; }
+
+import b2 = require('./b.json');
+>b2 : { [x: number]: number; }
+
+if (x) {
+>x : { [x: number]: number; }
+
+    x = b2;
+>x = b2 : { [x: number]: number; }
+>x : { [x: number]: number; }
+>b2 : { [x: number]: number; }
+}
+
+=== tests/cases/compiler/b.json ===
+{
+>{    [a]: 10} : { [x: number]: number; }
+
+    [a]: 10
+>[a] : number
+>a : any
+>10 : 10
+}

--- a/tests/cases/compiler/requireOfJsonFileWithComputedPropertyName.ts
+++ b/tests/cases/compiler/requireOfJsonFileWithComputedPropertyName.ts
@@ -1,0 +1,15 @@
+// @outdir: out/
+// @resolveJsonModule: true
+
+// @Filename: file1.ts
+import b1 = require('./b.json');
+let x = b1;
+import b2 = require('./b.json');
+if (x) {
+    x = b2;
+}
+
+// @Filename: b.json
+{
+    [a]: 10
+}

